src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -674,9 +696,13 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			this._cells.push(cell);
 			index = undefined;
 		}
-		// Set newly created cell as active cell
-		this.updateActiveCell(cell);
 		cell.isEditMode = true;
+		if (addToUndoStack) {
+			this.undoService.pushElement(new AddCellEdit(this, cell, index));
+			// Set newly created cell as active cell
+			this.updateActiveCell(cell);
```

Comment:
This should be outside the if check as well, right?

Reply:
If I update the active cell outside of this condition, it will make the cell active. I'm not sure if this will be the best experience for undoing  multiple times, because the user would have to deselect the cell first and then try to undo again.

Reply:
Gotcha, yeah that makes sense. Could you add that to the comment?